packages/ocs/dashboards/persistent-internal/encryption-popover.tsx

```diff
@@ -0,0 +1,235 @@
+import * as React from 'react';
+import { useODFNamespaceSelector } from '@odf/core/redux';
+import { KmsImplementations } from '@odf/core/types';
+import { ConfigMapKind, ConfigMapModel } from '@odf/shared';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { TFunction } from 'i18next';
+import {
+  Button,
+  Popover,
+  Text,
+  TextContent,
+  TextVariants,
+  Flex,
+  FlexItem,
+  Divider,
+} from '@patternfly/react-core';
+import { CheckCircleIcon, TimesCircleIcon } from '@patternfly/react-icons';
+
+type EncryptionPopoverProps = {
+  encryptionDetails: {
+    clusterWide?: boolean;
+    kms?: { enable: boolean };
+    storageClass?: boolean;
+  };
+  inTransitEncryption: boolean;
+  hasStorageClassEncryption: boolean;
+  hasMCGEncryption: boolean;
+};
+
+// React Component for rendering encryption status with icons
+const StatusIcon: React.FC<{ enabled: boolean }> = ({ enabled }) =>
+  enabled ? (
+    <CheckCircleIcon color="var(--pf-global--success-color--100)" />
+  ) : (
+    <TimesCircleIcon color="var(--pf-global--danger-color--100)" />
+  );
+
+// Utility function to get KMS type description from kmsConfig
+const getKmsTypeDescription = (encryptionKMSType: string, t: TFunction) => {
+  switch (encryptionKMSType) {
+    case KmsImplementations.VAULT_TENANT_SA:
+      return t('External Key Management Service: Vault');
+    case KmsImplementations.IBM_KEY_PROTECT:
+      return t(
+        'External Key Management Service: Hyper Protect Crypto Services'
+      );
+    case KmsImplementations.KMIP:
+      return t(
+        'External Key Management Service: Thales CipherTrust Manager (using KMIP)'
+      );
+    case KmsImplementations.AZURE:
+      return t('External Key Management Service: Azure Key Vault');
```

Comment:
no need to externalise KMS provider names...
create an enum of names under `odf/constants/kms.ts` and import from there and display....
```suggestion
      return t('External Key Management Service: Vault');
    case KmsImplementations.IBM_KEY_PROTECT:
      return t(
        'External Key Management Service: Hyper Protect Crypto Services'
      );
    case KmsImplementations.KMIP:
      return t(
        'External Key Management Service: Thales CipherTrust Manager (using KMIP)'
      );
    case KmsImplementations.AZURE:
      return t('External Key Management Service: {{ provider }}', {provider: <READ_FROM_ENUM> });
```

Reply:
Sure